### ISSUE TITLE
enhancement: Postponed wreg verif WCOM

### ISSUE THREAD

**Issue by CDPotter** [MN-Script-Team/DHS-MAXIS-Scripts#2123]
It has come to my attention that TIKLing to close for the verif due date may lead to cases staying open longer than needed. Per TE02.05.70.01 for cases where client's have used all their abawd months we need to close them on the appropriate date (to me this originally meant the 10 day return date) which appears to be close the case for the 1st of the 2nd month if application date is before 15, and 1st of 3rd month if application is post 15th.

I see a few solutions.

Creating a list of cutoff dates in FUNCLIB that we maintain
Creating something to calculate cutoff dates (could be tricky)
Changing the TIKL to occur based on application date (so if pre 15 then current month/day TIKL if post then next month on the 1st):
"Close case by cutoff this month for postponed verifications not returned(Allow to cutoff for client to turn them in)"

Thoughts?

**Comment by theVKC** [comment]
* Calculating cutoff dates is a non-starter, I think. Too complicated and error prone.
* Changing TIKL to occur based on application date could be error prone too, I'd think, because there are days where there is a shorter or longer 10-day period. I could see it getting confused.

I think it's time... 

`next_cutoff_date` could be a variable which determines itself based on a cutoff list on FuncLib. That's my vote. Maybe `current_mo_cutoff_date` and `next_mo_cutoff_date`, and we insert all known cutoff dates and update monthly/bimonthly. Thoughts?

**Comment by RobertFewins-Kalb** [comment]
I think absolutely yes to  next_cutoff_date being added to FuncLib. That information is available well in advance in SIR and could be easily updated.

**Comment by CDPotter** [comment]
Agreed, I think a list makes the most sense. I agree calculating is too error prone, and something we'd never catch or something went wrong.

**Comment by CDPotter** [comment]
Oh random thought! What about a function:

Function cutoff_date_calc(footer_month, cutoff_date)
  if footer_month = "04" THEN cutoff_date = "04/20/16" 
  ETC
end function

Though the draw back to this being more writer input/higher change to miss remembering to change it when the year changes.

**Comment by RobertFewins-Kalb** [comment]
/\ /\ /\ 
Yes please.

**Comment by theVKC** [comment]
@CDPotter: here's what I'm thinking...

1. We create a FuncLib issue, tie it to May release, and create the new function/constant first.
2. We update this issue description with a list of scripts that could benefit from this
3. We do the updating in step 2
4. Everyone is happy

Thoughts?

**Comment by CDPotter** [comment]
Sure that works for me. Want me to throw the issue up?

**Comment by theVKC** [comment]
Please do.

**Comment by theVKC** [comment]
Can I get an update on this @CDPotter? Do you think it'll be done by Friday (the due-date for the next release enhancements)? Thanks!

**Comment by CDPotter** [comment]
Waiting on cutoff constants, then I can start.

https://github.com/MN-Script-Team/BZS-FuncLib/pull/212

**Comment by theVKC** [comment]
I see. Want me to kick it to the June release?

**Comment by CDPotter** [comment]
Well if it gets pulled I can start work as early as Wednesday afternoon after my interviews are over, but if it's not getting pulled soon then we should bump it.

**Comment by theVKC** [comment]
@CDPotter, FYI we pulled those functions in. Good luck!
